frontend/src/container/Header.jsx

```diff
@@ -77,7 +77,7 @@ export class Header extends React.Component {
 
     const fetchPostUserLogout = await props.dispatch(postUserLogout())
     if (fetchPostUserLogout.status === 204) {
-      props.tlm.manager.closeLiveMessageConnection()
+      LiveMessageManager.getInstance().closeLiveMessageConnection()
```

Comment:
I'm not sure that introducing a singleton is a really good idea as it makes the code more difficult to test and restrict the used class to have only one instance (which is currently what it is, but we cannot totally be sure it will always be the case). See https://www.michaelsafyan.com/tech/design/patterns/singleton for an argumentation against singletons.

Did you d this to get rid of un-necessary refreshes of the application? If yes, couldn't we find a way to keep the object passed as props without those refreshes?

Reply:
Singleton is a bad design. The right way to implement this behavior is:
- implement a normal class
- implement a static global shared instance (can be implemented through a factory that manage a unique instance globally)

I'm pretty sure this pattern is used in backend code for an example

Reply:
@lebouquetin a factory has pretty much the same drawbacks as a singleton, of which I am aware (except that it forbids creating a new instance by construction). I had my reasons to go this way, you can't tell me this is bad design without asking why I did this.

You are both right though, it is best to avoid these patterns and after discussion with @grignards we found an elegant solution that solves the problem I am trying to  solve (avoid a bad refresh of the whole app when the connection goes live, which is completely unnecessary and breaks frontend tests), removes the need for a factory or a singleton, and makes the code cleaner and easier to work with: the best of every worlds.